File: deaf_grandma.js
function deafGrandma(str) {
    let abc = "abcdefghijklmnopqrstuvwxyz"
    if (str === "") {
    console.log("WHAT?!")
    } else if (str.length > 0) {
// let arr = []
// arr.push(str)
// let arr2 = arr[0].split('')
// return abc
for (let i=0; i<str.length; i++) {
    if (abc.includes(str[i])) {
        console.log("SPEAK UP, KID!")
    } else {console.log("NO, NOT SINCE 1946!")}}

if (str.includes("GOODBYE")) {
    console.log("LEAVING SO SOON?")
}}
// console.log(abc)
// if (str2.includes())
}

console.log(deafGrandma(''));
console.log(deafGrandma('hello'));
console.log(deafGrandma('HELLO'));
console.log(deafGrandma('GOODBYE'));
console.log(deafGrandma('GOODBYE, GOODBYE'));


// Write a program which can imitate a Grandma who's hard of hearing and follows these constraints:

// If you don't input anything (just hit enter) she responds with WHAT?!

// If you ask a question with any lower-case letters, she responds with SPEAK UP, KID!

// If you talk to her in all upper-case letters, she responds with NO, NOT SINCE 1946!

// The first time you say GOODBYE! she says LEAVING SO SOON?

// The second time you say GOODBYE! she says LATER, SKATER! and the program exits.

File: deaf_grandma.py
def deaf_grandma(g, str):
    
    # input('command Prompt')
    if len(str) > 0 and str != str.upper():
        print("SPEAK UP, KID!")
        deaf_grandma(0, input('command Prompt'))
    elif str == str.upper() and str != '' and str != "" and str != "GOODBYE!":
        print("NO, NOT SINCE 1946!")
        deaf_grandma(0, input('command Prompt'))
    elif str == "GOODBYE!" and g == 0:
        print("LEAVING SO SOON?")
        g = g + 1
        deaf_grandma(1, input('command Prompt'))
    elif str == "GOODBYE!" and g > 0:
        print("LATER, SKATER!")
    else:
        if len(str) == 0:
            print("WHAT?!")
            deaf_grandma(0, input('command Prompt'))
            # print("something went wrong")
            # input('command Prompt')
    

deaf_grandma(0, input('command Prompt'))

# deaf_grandma('')
# deaf_grandma('hi, grandma!')
# deaf_grandma('Hi, grandma!')
# deaf_grandma('I SAID HI, GRANDMA!')
# deaf_grandma('GOODBYE!')
# deaf_grandma('GOODBYE!')
# deaf_grandma('HELLO')


# HEY KID!
# > hi, grandma
# SPEAK UP, KID!
# > I SAID HI, GRANDMA
# NO, NOT SINCE 1946!
# >
# WHAT?!
# > Goodbye!
# SPEAK UP, KID!
# > GOODBYE!
# LEAVING SO SOON?
# > GOODBYE!
# LATER, SKATER!